src/tbb/arena.h

```diff
@@ -549,26 +573,37 @@ void arena::advertise_new_work() {
     }
 }
 
-inline d1::task* arena::steal_task(unsigned arena_index, FastRandom& frnd, execution_data_ext& ed, isolation_type isolation) {
-    auto slot_num_limit = my_limit.load(std::memory_order_relaxed);
-    if (slot_num_limit == 1) {
-        // No slots to steal from
-        return nullptr;
+/*  Trying to find index in the passed range using pool mask
+    Pre-condition: range.second <= my_limit */
+inline long long arena::get_index_for_steal_in_range(FastRandom& frnd, const std::pair<std::size_t, std::size_t>& range) {
```

Comment:
```suggestion
inline int arena::get_index_for_steal_in_range(FastRandom& frnd, const std::pair<std::size_t, std::size_t>& range) {
```
Why do you return `long long` if `try_steal_from_slot()` accepts the `slot_number` parameter as `int`?